pkg/apis/clusterapi/validation/cluster_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clusteroperatorapi "github.com/openshift/cluster-operator/pkg/api"
+	clusteroperator "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster"
+)
+
+func getValidClusterProviderConfig() *clusteroperator.ClusterProviderConfigSpec {
+	spec := &clusteroperator.ClusterProviderConfigSpec{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterProviderConfigSpec",
+			APIVersion: "clusteroperator.openshift.io/v1alpha1",
+		},
+		ClusterSpec: clusteroperator.ClusterSpec{
+			ClusterVersionRef: clusteroperator.ClusterVersionReference{
+				Name: "someclusterref",
+			},
+			MachineSets: []clusteroperator.ClusterMachineSet{
+				{
+					ShortName: "master",
+					MachineSetConfig: clusteroperator.MachineSetConfig{
+						NodeType: "Master",
+						Size:     1,
+					},
+				},
+				{
+					ShortName: "infra",
+					MachineSetConfig: clusteroperator.MachineSetConfig{
+						NodeType: "Compute",
+						Infra:    true,
+						Size:     1,
+					},
+				},
+			},
+		},
+	}
+	return spec
+}
+
+func getValidClusterWithEmptyProviderConfig() *cluster.Cluster {
+	cluster := &cluster.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster",
+		},
+		Spec: cluster.ClusterSpec{
+			ProviderConfig: cluster.ProviderConfig{
+				Value: nil,
+			},
+		},
+	}
+	return cluster
+}
+
+func getRawExtensionFromProviderConfig(providerConfig *clusteroperator.ClusterProviderConfigSpec) *runtime.RawExtension {
+	serializer := jsonserializer.NewSerializer(jsonserializer.DefaultMetaFactory, clusteroperatorapi.Scheme, clusteroperatorapi.Scheme, false)
+	var buffer bytes.Buffer
+	serializer.Encode(providerConfig, &buffer)
```

Comment:
Don't swallow the error returned from `Encode`. It will make it more difficult to debug if there is a problem encoding.